chemprop/cli/train.py

```diff
@@ -636,6 +639,83 @@ def build_splits(args, format_kwargs, featurization_kwargs):
 
 
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    if args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+        y_mean = np.nanmean(y, axis=0)
+        print("YMEAN", y_mean)
+        y_std = np.nanstd(y, axis=0)
+        y_median = np.nanmedian(y, axis=0)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / len(y)
+        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / len(y)
+
+        column_headers = ["Statistic"] + [f"Value {i}" for i in range(y.shape[1])]
+        table_rows = [
+            ["Num. datapoints"] + [f"{len(y)}" for i in range(y.shape[1])],
+            ["Mean"] + [f"{mean:0.2f}" for mean in y_mean],
+            ["Std. dev."] + [f"{std:0.2f}" for std in y_std],
+            ["Median"] + [f"{median:0.2f}" for median in y_median],
+            ["% within 1 s.d."] + [f"{sigma:0.0%}" for sigma in frac_1_sigma],
+            ["% within 2 s.d."] + [f"{sigma:0.0%}" for sigma in frac_2_sigma],
+        ]
+        return (column_headers, table_rows)
+    elif args.task_type in [
+        "classification",
+        "classification-dirichlet",
+        "multiclass",
+        "multiclass-dirichlet",
+    ]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+
+        mask = np.isnan(y)
+        classes = np.sort(np.unique(y[~mask]))
+
+        class_counts = np.stack([(classes[:, None] == y[:, i]).sum(1) for i in range(y.shape[1])])
+        class_fracs = class_counts / y.shape[0]
+        nan_count = np.nansum(mask, axis=0)
+        nan_frac = nan_count / len(y)
+
+        column_headers = ["Class"] + [f"Count/Percent {i}" for i in range(y.shape[1])]
+
+        table_rows = []
+        for i, k in enumerate(classes):
+            class_row = [f"{k}"] + [
+                f"{class_counts[j,i]}/{class_fracs[j,i]:0.0%}" for j in range(y.shape[1])
+            ]
+            table_rows.append(class_row)
+
+        nan_row = ["NAN"] + [f"{nan_count[i]}/{nan_frac[i]:0.0%}" for i in range(y.shape[1])]
+        table_rows.append(nan_row)
+
+        total_row = ["Total"] + [f"{y.shape[0]}/{100.00}%" for i in range(y.shape[1])]
+        table_rows.append(total_row)
+
+        return (column_headers, table_rows)
+    else:
+        raise ValueError(f"unsupported task type! Task type '{args.task_type}' was not recognized.")
+
+
+def build_table(column_headers: list[str], table_rows: list[str], title: str | None = None) -> str:
+    right_justified_columns = [
+        Column(header=column_header, justify="right") for column_header in column_headers
+    ]
+    table = Table(*right_justified_columns, title=title)
+    for row in table_rows:
+        table.add_row(*row)
+    from io import StringIO
```

Comment:
I think others on the team would suggest you move the import up to the top of the file with the other imports. I don't care too much myself.

Reply:
Done!